clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -543,6 +543,12 @@ void SYCLToolChain::AddSYCLIncludeArgs(const clang::driver::Driver &Driver,
   CC1Args.push_back(DriverArgs.MakeArgString(P));
 }
 
+void SYCLToolChain::AddDevicelibIncludeArgs(ArgStringList &CC1Args) {
+  // Assume that clang system include path is added elsewhere
+  CC1Args.push_back("-include");
+  CC1Args.push_back("devicelib/__devicelib_assert.h");
```

Comment:
Can this be included somewhere in SYCL headers?

Reply:
The point is to support compilation when SYCL headers are not included, e.g. when SYCL_EXTERNAL is used.

Reply:
+1. 
The idea here is **the compiler** to provide additional functionality for SPIR target through the functions declared in this header.
Current implementation is not quite implements this idea - it includes the header for SYCL mode rather then SPIR target, so we probably have to skip this include for non-SPIR targets (e.g. nvidia).

Reply:
> The point is to support compilation when SYCL headers are not included

Compilation of what?

Reply:
> > The point is to support compilation when SYCL headers are not included
> 
> Compilation of what?

Compilation of source code that do not `#include` SYCL headers.

Reply:
> Current implementation is not quite implements this idea - it includes the header for SYCL mode rather then SPIR target, so we probably have to skip this include for non-SPIR targets (e.g. nvidia).

The last patch includes the header when SYCL language is compiled for SPIR target. Non-SPIR targets should not be affected.

Reply:
Andrew, can we make this include agnostic to compilation mode/language and include it for all compilations for SPIR target?

Reply:
If it makes sense - sure. I'm concerned that this feature will be enabled for languages where it doesn't make sense (for OpenCL, in particular).

Reply:
Why do you think it doesn't make sense for OpenCL?